subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/ScriptHandlerScope.kt

```diff
@@ -46,8 +47,8 @@ class ScriptHandlerScope(scriptHandler: ScriptHandler) : ScriptHandler by script
     /**
      * The script classpath configuration.
      */
-    val ConfigurationContainer.classpath: Configuration
-        get() = getByName(CLASSPATH_CONFIGURATION)
+    val NamedDomainObjectContainer<Configuration>.classpath: NamedDomainObjectProvider<Configuration>
```

Comment:
:x: Why?

Reply:
Because `configurations {}` receiver is `NamedDomainObjectContainerScope<Configuration>`.